Buildings/Fluid/HeatExchangers/BaseClasses/WetCoilUARated.mo

```diff
@@ -0,0 +1,192 @@
+
+  replaceable package MediumA=Buildings.Media.Air
+    constrainedby Modelica.Media.Interfaces.PartialMedium
+    "Air-side medium";
+  replaceable package MediumW=Buildings.Media.Water
+    constrainedby Modelica.Media.Interfaces.PartialMedium
+    "Water-side medium";
+
+  parameter Boolean use_UA_nominal = false
+    "Set to true to specify UA_nominal, or to false to use nominal conditions"
+    annotation (Evaluate=true,
+                Dialog(group="Nominal thermal performance"));
+  parameter Modelica.SIunits.Temperature TAirIn
+    "Air inlet temperature at a rated condition";
+  parameter Modelica.SIunits.Temperature TAirOut
+    "Air outlet temperature  at a rated condition";
+  parameter Modelica.SIunits.MassFraction X_wAirIn
+    "Mass fraction of water in inlet air at a rated condition";
+  parameter Modelica.SIunits.Temperature TWatIn
+    "Water inlet temperature at a rated condition";
+  parameter Modelica.SIunits.Temperature TWatOut
+    "Water outlet temperature at a rated condition";
+  parameter Modelica.SIunits.MassFlowRate mAir_flow
+    "Air mass flow rate at a rated condition";
+  parameter Modelica.SIunits.MassFlowRate mWat_flow
+    "Water mass flow rate at a rated condition";
+  parameter Modelica.SIunits.ThermalConductance UA
+    "UA is for the overall coil (i.e., both sides)";
+  parameter Real r_nominal(min=0, max=1)
+    "Ratio between air-side and water-side convective heat transfer at nominal condition";
+protected
+  constant Modelica.SIunits.SpecificEnthalpy hfg=
+    Buildings.Utilities.Psychrometrics.Constants.h_fg;
+  parameter Modelica.SIunits.MassFraction X_wAirOut(fixed=false)
+    "Mass fraction of water in outgoing air at a rated condition";
+  parameter Modelica.SIunits.SpecificEnthalpy hAirIn(fixed=false)
+    "Enthalpy of incoming moist air at a rated condition";
+  parameter Modelica.SIunits.SpecificEnthalpy hAirOut(fixed=false)
+    "Enthalpy of outgoing moist air at a rated condition";
+
+  // calculates all thermodynamic properties based on inputs
+  parameter MediumA.ThermodynamicState staAir=MediumA.setState_phX(
+    p=MediumA.p_default,h=MediumA.h_default,X=MediumA.X_default[1:MediumA.nXi]);
+  parameter MediumW.ThermodynamicState staWat=MediumW.setState_phX(
+    p=MediumW.p_default,h=MediumW.h_default,X=MediumW.X_default[1:MediumW.nXi]);
+
+  parameter Modelica.SIunits.SpecificHeatCapacity cpAir=
+    MediumA.specificHeatCapacityCp(staAir);
+  parameter Modelica.SIunits.SpecificHeatCapacity cpw=
+    MediumW.specificHeatCapacityCp(staWat);
+  parameter Modelica.SIunits.SpecificHeatCapacity cpEff(fixed=false, min= 0)
+    "Effective specific heat: change in saturated moist air enthalpy with respect to
+    temperature along the saturation line between inlet and outlet water temperatures";
+
+  constant Modelica.SIunits.SpecificEnthalpy hunit=1;
+  constant Modelica.SIunits.Temperature Tunit=1;
+  constant Modelica.SIunits.SpecificHeatCapacity cpunit=1;
+
```

Comment:
All parameters, constants and variables must have a documentation string.

Reply:
Revised.